test/test_cuda.py

```diff
@@ -4884,6 +4884,27 @@ def test_no_triton_on_import(self):
 
 @unittest.skipIf(not TEST_CUDA, "CUDA not available, skipping tests")
 class TestMemPool(TestCase):
+    def _setup_mempool_limited_memory_test(self, additional_allowed_memory_in_mb):
+        os.environ["PYTORCH_CUDA_ALLOC_CONF"] = "expandable_segments:True"
+
+        device = torch.device("cuda:0")
+
+        torch.cuda.memory.empty_cache()
+        mb = 1024 * 1024
+        _, all_memory = torch.cuda.memory.mem_get_info(device)
+        pre_reserved = torch.cuda.memory_reserved(device)
+        total_allowed = additional_allowed_memory_in_mb * mb + pre_reserved
+        fraction_allowed = total_allowed / all_memory
+        torch.cuda.memory.set_per_process_memory_fraction(fraction_allowed, device)
+
+        dtype = torch.int8
+        return device, dtype
+
+    def _teardown_mempool_limited_memory_test(self):
+        os.environ["PYTORCH_CUDA_ALLOC_CONF"] = ""
```

Comment:
See above, I'm not sure we expect `PYTORCH_CUDA_ALLOC_CONF` to be re-parsed every time? https://github.com/pytorch/pytorch/blob/81c4369d813facf39313dfd481adc71704cbc2c1/c10/cuda/CUDAAllocatorConfig.h#L81